crates/rome_rowan/src/ast/mod.rs

```diff
@@ -364,6 +388,57 @@ impl<L: Language, N: AstNode<Language = L>> FusedIterator
 {
 }
 
+impl<L: Language, N: AstNode<Language = L>> DoubleEndedIterator
+    for AstSeparatedListElementsIterator<L, N>
+{
+    fn next_back(&mut self) -> Option<Self::Item> {
+        let slot = self.slots.next_back();
```

Comment:
You can use the `try` operator here so that the implementation returns immediately if you reached the start of the list
```suggestion
        let slot = self.slots.next_back()?;
```

Reply:
I can't here, because I might be at the end of the list where the separator isn't there and we have a node instead: `vec![Some(1), Some(","), Some(1)]`.